backend/src/apiserver/storage/run_store.go

```diff
@@ -238,12 +260,16 @@ func parseResourceReferences(resourceRefString sql.NullString) ([]*model.Resourc
 }
 
 func (s *RunStore) CreateRun(r *model.RunDetail) (*model.RunDetail, error) {
+	if r.StorageState == "" {
```

Comment:
But this would put the burden on clients to fill in a value for that field. I think this isn't something clients should be forced to do, this is why I also removed the UNSPECIFIED enum option.
wdyt?

Reply:
Sync'ed offline. Added some validation around this, PTAL.